src/Console/Command/DescribeCommand.php

```diff
@@ -265,7 +272,8 @@ private function describeSet(OutputInterface $output, $name)
             throw new DescribeNameNotFoundException($name, 'set');
         }
 
-        $ruleSet = new RuleSet(array($name => true));
+        $ruleSetClass = get_class($this->ruleSet);
+        $ruleSet = new $ruleSetClass(array($name => true));
```

Comment:
>and more, it's wrong, as constructor accept any interface, so given class could have private constructor

The interface https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.3/src/RuleSetInterface.php#L24 forces the constructor to be public

What _is_ messy; is the `RuleSet` being both an instance of a collection of rules and a collection of _all_ sets, which makes this needed.
I.e we do _not_ allow someone to create a `RuleSet` of their own and add it to the set of all `RuleSet`'s as the later does not exists.